fineract-provider/src/integrationTest/java/org/apache/fineract/integrationtests/common/SchedulerJobHelper.java

```diff
@@ -122,45 +139,112 @@ private static String runSchedulerJobAsJSON() {
 
-                // print error associated with recent job failure (if any)
-                System.out.println("Job run error message (printed only if the job fails: "
-                        + jobHistoryData.get(jobHistoryData.size() - 1).get("jobRunErrorMessage"));
-                System.out.println("Job failure error log (printed only if the job fails: "
-                        + jobHistoryData.get(jobHistoryData.size() - 1).get("jobRunErrorLog"));
+        assertFalse("Job History is empty :(  Was it too slow? Failures in background job?", jobHistoryData.isEmpty());
 
-                // Verifying the Status of the Recently executed Scheduler Job
-                Assert.assertEquals("Verifying Last Scheduler Job Status", "success",
-                        jobHistoryData.get(jobHistoryData.size() - 1).get("status"));
+        // print error associated with recent job failure (if any)
+        System.out.println("Job run error message (printed only if the job fails: "
+                + jobHistoryData.get(jobHistoryData.size() - 1).get("jobRunErrorMessage"));
+        System.out.println("Job failure error log (printed only if the job fails: "
+                + jobHistoryData.get(jobHistoryData.size() - 1).get("jobRunErrorLog"));
+
+        // Verifying the Status of the Recently executed Scheduler Job
+        assertEquals("Verifying Last Scheduler Job Status", "success",
+                jobHistoryData.get(jobHistoryData.size() - 1).get("status"));
+    }
 
-                break;
+    /**
+     * Launches a Job and awaits its completion.
+     * @param jobName displayName (see {@link org.apache.fineract.infrastructure.jobs.service.JobName}) of Scheduler Job
+     *
+     * @author Michael Vorburger.ch
+     */
+    public void executeAndAwaitJob(String jobName) {
+        Duration TIMEOUT = Duration.ofSeconds(30);
+        Duration PAUSE = Duration.ofMillis(500);
+        DateTimeFormatter df = DateTimeFormatter.ISO_INSTANT; // FINERACT-926
+        Instant beforeExecuteTime = now().truncatedTo(ChronoUnit.SECONDS);
+
+        // Stop the Scheduler while we manually trigger execution of job, to avoid side effects and simplify debugging when readings logs
+        updateSchedulerStatus(false);
+
+        // Executing Scheduler Job
+        int jobId = getSchedulerJobIdByName(jobName);
+        runSchedulerJob(jobId);
+
+        // Await JobDetailData.lastRunHistory [JobDetailHistoryData] jobRunStartTime >= beforeExecuteTime (or timeout)
+        await().atMost(TIMEOUT).pollInterval(PAUSE).until(jobLastRunHistorySupplier(jobId), lastRunHistory -> {
+            String jobRunStartText = lastRunHistory.get("jobRunStartTime");
+            if (jobRunStartText == null) {
+                return false;
             }
+            Instant jobRunStartTime = df.parse(jobRunStartText, Instant::from);
+            return jobRunStartTime.equals(jobRunStartTime) || jobRunStartTime.isAfter(beforeExecuteTime);
```

Comment:
@ptuomola thank you for having taken the time for a "real" (full, detailed - as opposed to just "waving through") code review - I appreciate that very much! So yeah testing that jobRunStartTime is equal to itself was definitely wrong here. But I don't think we want to beforeExecuteTime and jobRunStartTime either - that should always be the case (job should always have started just on or later than the test start). What I really wanted to do here is `jobRunEndTime.equals(jobRunStartTime) || jobRunEndTime.isAfter(jobRunStartTime)` - makes sense? Now so done. Please do ask if unclear!

Reply:
@vorburger I think the second check (row 228) is correct as you state (i.e. jobRunEndTime.equals(jobRunStartTime) || jobRunEndTime.isAfter(jobRunStartTime))

But the first check I was commenting on (row 217) would still make most sense to me as:

jobRunStartTime.equals(beforeExecuteTime) || jobRunStartTime.isAfter(beforeExecuteTime)

In order to make sure that the job has had a chance to start, before we start looking at its history. 

If we don't do this and go straight to comparing jobRunStartTime and jobRunEndTime, I think there's a risk of hitting the same issue as I observed earlier: i.e. that we started looking for the history before the job even had had a chance to start for this test. 

In such case we would get the previous execution history of the job, jobRunEndTime would still be >= jobRunStartTime - but this would be the previous run of the job, not the current one we had triggered.

Or am I missing something here?

Reply:
No you're not missing anything, I was just looking at the wrong line! Yes, absolutely agree - better now? However this actually now reveals a new real bug! See FINERACT-992... do you happen to have any interest in taking that? Just asking! I'm marking this PR Draft again after all, as we probably have to fix that issue before this can be merged.

Reply:
@ptuomola I'm resolving this because I have just closed  FINERACT-992. 

Please re-open if NOK for you.